### DISCUSSION TITLE
Argentina API

### DISCUSSION THREAD

**Discussion by jmcarranza** [electricitymaps/electricitymaps-contrib#3307] [Ideas]
Hi, maybe you can try https://api.cammesa.com/demanda-svc/swagger-ui.html# for more accurate data.

**Comment by jmcarranza** [comment]
more public data https://api.cammesa.com/pub-svc/swagger-ui.html#/

**Comment by madsnedergaard** [comment]
Thanks for sharing :) 
Argentina is already [on the map](https://www.electricitymap.org/zone/AR), do you think that it's missing some data?

**Reply by jmcarranza** [reply]
Hi, The scrapper is good but is not up to date.
The api is more accurate and up to date on any change of the generation and demand.

**Reply by madsnedergaard** [reply]
Ah okay, now I see - if you or anyone have time to work on this that would be great!

I assume the data we find on the website is coming from the same API, so not sure how much the result would actually differ though? :)

**Reply by jmcarranza** [reply]
the data you are scraping may soon be unavailable. The API is official and the information is completely updated.

**Comment by jarek** [comment]
Oh that's nice! Copying in a sample datapoint from `curl -X GET "https://api.cammesa.com/demanda-svc/generacion/ObtieneGeneracioEnergiaPorRegion?id_region=1002" -H  "accept: */*"` for discussion:

```
  {
    "fecha": "2021-08-21T00:10:00.000-0300",
    "sumTotal": 19226.2,
    "hidraulico": 3105.4,
    "termico": 13512.5,
    "nuclear": 1598.5,
    "renovable": 1009.8
  },
```

One downside I see is that this seems to group all "thermal" generation together, whereas with the current parser (when it works) we split out natural gas (methane), oil, and coal. It's not that big of a deal since it looks like vast majority of generation is from gas, so we could have "thermal" in "unknown" and adjust its carbon intensity to match capacity ratio or historic generation ratio of the different fuels.

"Renewables" are also grouped together whereas the current parser gets wind and solar separately. Electricity Map can't really deal with a generic "renewables" source very well. Considering the current generation numbers, classifying it all as wind (and maybe bumping up the carbon intensity a bit to account for the solar part) might be defensible, but it seems inelegant...
